src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_SIZE = 100;
+    private static final int MIN_SIZE = 1;
+    private static final int FIGURES_NUMBER = 4;
+
+    public Figure getRandomFigure() {
+        Random random = new Random();
+        String color = new ColorSupplier().getRandomColor();
+        switch (random.nextInt(FIGURES_NUMBER)) {
+            case 0:
+                return new Rectangle(random.nextInt(MAX_SIZE) + MIN_SIZE, random.nextInt(MAX_SIZE)
+                        + MIN_SIZE, color);
```

Comment:
don't pass expressions in constructor arguments. let's create separate variables and pass them.  and to keep code clean do all of these in a private method.